node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -279,7 +279,20 @@ impl OrderingProvider {
 				rx.await.or(Err(Fatal::ChainApiSenderDropped))?.map_err(Fatal::ChainApi)?
 			};
 
-			let earliest_block_number = head_number - hashes.len() as u32;
+			let earliest_block_number = match head_number.checked_sub(hashes.len() as u32) {
+				Some(number) => number,
+				None => {
+					// It's assumed that it's impossible to retrieve
+					// more than N ancestors for block number N.
+					tracing::error!(
+						target: LOG_TARGET,
+						"Received {} ancestors for block number {} from Chain API",
+						hashes.len(),
+						head_number,
+					);
+					return Ok(ancestors)
+				},
+			};
```

Comment:
Doing `saturated_sub` is bad since we will have invalid block number for the next iteration in case of underflow, so it's best to assume this is unreachable.